model/src/pyrenew/metaclass.py

```diff
@@ -721,6 +723,12 @@ def sample(self, **kwargs) -> tuple:
             for t, uv in zip(self.transforms, untransformed_values)
         )
 
+        if record:
+            for i, tv in enumerate(transformed_values):
+                numpyro.deterministic(f"{self.name}_{i}", tv.value)
```

Comment:
If the original `RandomVariable` `sample` is a `namedtuple`, can we re-use the name from the `namedtuple`?

Reply:
Do you mean use the same name as the untransformed `RandomVariable` instead of the `name` argument for the `TransformedRandonVariable`?

Reply:
It's a bit confusing. `RandomVariable`s do not necessarily have a `name`.

`RandomVariable.sample()` always returns a tuple of `SampleValue`s. This tuple is sometimes a namedtuple.  This is the tuple that you are looping over in the code we are commenting on. I am wondering if we can use the names in the tuple as suffixes for the name in the `numpyro.deterministic` call.

Reply:
Ok, you mean use the `base_rv` name as a suffix! Will try